pkg/kv/kvserver/replica_init.go

```diff
@@ -229,19 +291,6 @@ func (r *Replica) loadRaftMuLockedReplicaMuLocked(desc *roachpb.RangeDescriptor)
 	}
 
 	r.setDescLockedRaftMuLocked(ctx, desc)
-
-	// Only do this if there was a previous lease. This shouldn't be important
-	// to do but consider that the first lease which is obtained is back-dated
-	// to a zero start timestamp (and this de-flakes some tests). If we set the
-	// min proposed TS here, this lease could not be renewed (by the semantics
-	// of minLeaseProposedTS); and since minLeaseProposedTS is copied on splits,
-	// this problem would multiply to a number of replicas at cluster bootstrap.
-	// Instead, we make the first lease special (which is OK) and the problem
-	// disappears.
-	if r.mu.state.Lease.Sequence > 0 {
-		r.mu.minLeaseProposedTS = r.Clock().NowAsClockTimestamp()
-	}
```

Comment:
Copied this to `newReplica`. It is ok to not leave it here, because the only user of `initRaftMuLockedReplicaMuLocked` is the split trigger, and it rewrites `r.mu.minLeaseProposedTS` right after calling this.

The split trigger essentially does more post-initialization things, which might need to be revisited and included here.